lib/js/components/CounterToggle/CounterToggle.vue

```diff
@@ -0,0 +1,377 @@
+			'background': $color-primary-background,
+			'background-hovered': $color-primary-background-hovered,
+			'background-focused': $color-primary-background-focused,
+			'background-pressed': $color-primary-background-hovered,
+			'icon': $color-primary-icon,
+			'ripple': $color-primary-ripple,
+			'disabled': (
+				'color': $color-primary-text-disabled,
+				'icon': $color-primary-icon-disabled,
+				'background': $color-primary-background-disabled,
+			),
+		),
+	),
+);
+
+@mixin setCounterToggleColor(
+	$color,
+	$color-hovered: null,
+	$color-focused: null,
+	$color-disabled: null
+) {
+	color: $color;
+
+	&:hover,
+	&.-hovered,
+	&:active,
+	&.-active {
+		color: $color-hovered;
+	}
+
+	&:focus,
+	&.-focused {
+		color: $color-focused;
+	}
+
+	&.-disabled {
+		color: $color-disabled;
+	}
+}
+
+@mixin setCounterToggleBackground(
+	$background,
+	$background-hover,
+	$background-focus,
+	$background-pressed
+) {
+	@if $background == null {
+		background-color: transparent;
+	}
+
+	background-color: $background;
+
+	&:hover,
+	&.-hovered {
+		background-color: $background-hover;
+	}
+
+	&:active,
+	&.-active {
+		background-color: $background-pressed;
+	}
+
+	&:focus,
+	&.-focused {
+		background-color: $background-focus;
+	}
+}
+
+@mixin setCounterToggleDisabled(
+	$disabled-color: null,
+	$disabled-icon: null,
+	$disabled-background: null,
+	$disabled-border: null
+) {
+	&:disabled,
+	&.-disabled {
+		border-color: $disabled-border;
+		color: $disabled-color;
+		background-color: $disabled-background;
+
+		.counterToggle__icon {
+			color: $disabled-icon;
+		}
+	}
+}
+
+@mixin setCounterToggleAdditions($ripple: null, $icon: null) {
+	@if $ripple == null {
+		&::v-deep .ripple {
+			display: none;
+		}
+	}
+
+	&::v-deep .ripple {
+		background-color: $ripple !important;
+	}
+
+	.counterToggle {
+		&__icon {
+			color: $icon;
+		}
+	}
+}
+
+.counterToggle {
+	@each $color-name, $color-map in $counter-toggle-colors {
+		&.-color-#{$color-name} {
+			@include setCounterToggleColor(map-get($color-map, 'default', 'color'));
+			@include setCounterToggleBackground(
+				map-get($color-map, 'default', 'background'),
+				map-get($color-map, 'default', 'background-hovered'),
+				map-get($color-map, 'default', 'background-focused'),
+				map-get($color-map, 'default', 'background-pressed')
+			);
+			@include setCounterToggleDisabled(
+				map-get($color-map, 'default', 'disabled', 'color'),
+				map-get($color-map, 'default', 'disabled', 'icon'),
+				map-get($color-map, 'default', 'disabled', 'background')
+			);
+			@include setCounterToggleAdditions(
+				map-get($color-map, 'default', 'ripple'),
+				map-get($color-map, 'default', 'icon')
+			);
+		}
+	}
+
+	align-items: center;
+	background-color: transparent;
+	border: 0;
+	border-radius: $radius-s;
+	cursor: pointer;
+	display: inline-flex;
+	width: auto;
+	justify-content: center;
+	padding: $space-xxs;
+	position: relative;
+	text-transform: uppercase;
+	transform-style: preserve-3d;
```

Comment:
I don't see the difference with or without it, and I don't see any 3d transitions in this component. I think it was added to Button (which this is copied from) because of:
https://github.com/bethinkpl/design-system/blob/b5fb1bfd3cc51cb25f902a949aa68a48751b44de/lib/styles/components/_buttons.scss#L307

I'd remove it from here.